### ISSUE TITLE
String parameter cannot pass control characters (\n)

### ISSUE THREAD

**Issue by NpDen4ik** [DarkWanderer/ClickHouse.Client#117]
Hi i can't pass string with control character by parameter.

## Example:

```csharp
[Test]
public async Task ShouldInsertStringWithSpecialSymbols()
{
    await connection.ExecuteStatementAsync("TRUNCATE TABLE IF EXISTS test.string_with_special_symbols");
    await connection.ExecuteStatementAsync(
        "CREATE TABLE IF NOT EXISTS test.string_with_special_symbols (str_value String) ENGINE TinyLog");

    var command = connection.CreateCommand();

    var strValue = "Hello \n ClickHouse";
    
    command.AddParameter("str_value", strValue);
    command.CommandText = "INSERT INTO test.string_with_special_symbols VALUES ({str_value:String})";
    await command.ExecuteNonQueryAsync();

    var count = await connection.ExecuteScalarAsync("SELECT COUNT(*) FROM test.string_with_special_symbols");
    Assert.AreEqual(1, count);
}
```

## Exception:
```
Code: 457. DB::Exception: Value Hello 
 ClickHouse cannot be parsed as String for query parameter 'str_value' because it isn't parsed completely: only 6 of 18 bytes was parsed: Hello : While executing ValuesBlockInputFormat. (BAD_QUERY_PARAMETER) (version 22.1.2.2 (official build))
```

But it works in play.
![изображение](https://user-images.githubusercontent.com/32520717/150658115-82f16701-f1ea-4b3d-a48f-f1a7d694ca89.png)

Check by datagrip:
![изображение](https://user-images.githubusercontent.com/32520717/150658165-1c823058-041a-4223-9f1f-128c0de5252d.png)

---

## Versions
```
<PackageReference Include="ClickHouse.Client" Version="4.1.0" />
```
```
select version();
-- 22.1.2.2
```

**Comment by NpDen4ik** [comment]
For pass like inline parameter to **CommandText** you need formatted string content => replace \'

## Example:
```csharp
public static string FormattedStringForPassToCommand(string? value)
{
    return string.IsNullOrEmpty(value)
        ? ""
        : value.Replace("'", "\\'");
}
```

**Comment by DarkWanderer** [comment]
Released in [4.2.0](https://github.com/DarkWanderer/ClickHouse.Client/releases/tag/4.2.0)
